ros/src/sensing/drivers/camera/packages/vectacam/nodes/vectacam/VectaCam.cpp

```diff
@@ -0,0 +1,260 @@
+		return;
+	}
+	memset((char *) &socket_address, 0, sizeof(socket_address));
+	socket_address.sin_family = AF_INET;
+	socket_address.sin_port = htons(in_port);
+	if (inet_aton(CAMERA_IP, &socket_address.sin_addr)==0)
+	{
+		std::cout << "Invalid IP address" << std::endl;
+		return;
+	}
+	if (sendto(socket_descriptor, enable_command, sizeof(enable_command), 0, reinterpret_cast<sockaddr *>(&socket_address), sizeof(socket_address)) < 0)
+	{
+		std::cout << "Could not send the enable command to the camera socket" << std::endl;
+	}
+}
+
+void VectaCam::_initialize_camera(unsigned int in_configuration_port, unsigned int in_data_port, std::string in_parameter_file)
+{
+	std::cout << "Reading configuration file...";
+	_parse_parameter_file(parameter_file_, camera_commands_);
+	std::cout << "DONE" << std::endl << "Configuring camera.";
+	_send_commands_to_camera(in_configuration_port, camera_commands_);
+	std::cout << "DONE" << std::endl << "Enabling camera.";
+	_enable_camera(in_data_port, true);
+	std::cout << "DONE" << std::endl;
+}
+
+VectaCam::~VectaCam()
+{
+	this->running_=false;
+	delete[] this->image_buffer_;
+}
+
+void VectaCam::GetImage(cv::Mat& out_image)
+{
+	out_image = camera_image_.clone();
+}
+
+void VectaCam::StopCamera()
+{
+	this->running_ = false;
+	_enable_camera(data_port_, false);
+}
+
+long int VectaCam::GetFrameNumber()
+{
+	return this->frame_number_;
+}
+
+void VectaCam::StartCamera()
+{
+
+	int 		socket_descriptor;
+	struct 		sockaddr_in socket_address;
+	unsigned int length;
+
+	if ((socket_descriptor = socket( PF_INET, SOCK_DGRAM, 0)) < 0)
+	{
+		std::cout << "Problem creating socket\n";
+		return;
+	}
+
+	socket_address.sin_family = AF_INET;
+	socket_address.sin_addr.s_addr = htonl(INADDR_ANY);
+	socket_address.sin_port = htons(this->data_port_);
+
+	if (bind(socket_descriptor, reinterpret_cast<sockaddr *> (&socket_address), sizeof(socket_address)) < 0)
+	{
+		std::cout << "Problem binding" << std::endl;
+		return;
+	}
+
+	length = sizeof(socket_address);
+	if (getsockname(socket_descriptor, reinterpret_cast<sockaddr *> (&socket_address), &length) < 0)
+	{
+		std::cout << "Error getsockname" << std::endl;
+		return;
+	}
+	this->camera_ready_ = true;
+	this->running_ = true;
+
+	std::cout << "Listening camera in UDP port number " << std::to_string(ntohs(socket_address.sin_port)) << std::endl;
+
+	_udp_receive(socket_descriptor);
+}
+
+void VectaCam::_udp_receive(int in_socket_descriptor)
+{
+	unsigned int 	length, n;
+	char 			buffer_in[MAX_BUFFER];
+	struct 			sockaddr_in remote;
+	length = sizeof(remote);
+
+	int line_number = 0;
+
+	while (this->running_)
+	{
+		n = recvfrom(in_socket_descriptor, buffer_in, MAX_BUFFER, 0, reinterpret_cast<sockaddr *> (&remote), &length);
+		if (n < 0)
+		{
+			std::cout << "Error receiving data" << std::endl;
+		}
+		else
+		{
+			int packet_offset = ntohl(*(uint32_t*)(buffer_in + 4));
+			int packet_length = ntohl(*(uint32_t*)(buffer_in + 8));
```

Comment:
`packet_offset` and `packet_length` should be `uint32_t`. Please see `man ntohl`.